### DISCUSSION TITLE
open transparent remote / other backends files over file chooser

### DISCUSSION THREAD

**Discussion by spiderbit** [davatorium/rofi#1837] [Ideas/Feature Requests]
I am used to emacs TRAMP, so when I want to open let's say fstab on my server, I can open the file chooser add /ssh:root@server:/ then tab-complete to /etc/fstab open it, edit it, and save it.

Now the parser is not as important I would be ok to use something like the ssh port to open it, but more the need to not mount it manually and open with a local application a file here, I think it could be done by do a mount to /tmp/<generate-random-name>/ (before mkdir, or there is a function that does maybe both at the same time), and see the files from there, maybe over gvfs maybe over sshfs.

The same could be done over adb, samba, davs and many other protocols, it could just cache the files over ssh and have some size limit setup. The problem with caching is that nothing tracks then after opening that it's supposed to be uploaded again, and the apps would not show the remote path, therefor I think dynamic mounting would be the next best option.

Also maybe a simpler version would be to have a mount command that scans known_hosts and gvfs available stuff to mount, and then maybe another file command that shows a list of all mounted filesystems and then open files over that.

**Comment by DaveDavenport** [comment]
Not sure I follow this story or what the question/request is?

But opening/listing remote filesystems is very far out of the scope of rofi.
(but nothing is stopping your to implement a script/plugin that does what you want).
